### ISSUE TITLE
Site running PHP 5.6 throws error message "Use of iconv.input_encoding is deprecated" upon login attempt

### ISSUE THREAD

**Issue by rwetzlmayr** [SocialEngine/phpv4-issues#17]
I am attempting to install and run SE PHP 4.8.7 on a host running PHP 5.6.

Installation has commenced without any hiccups, but upon first login I am greeted with these messages:

```
Deprecated: iconv_set_encoding(): Use of iconv.input_encoding is deprecated in ../application/libraries/Engine/Application.php on line 314
Deprecated: iconv_set_encoding(): Use of iconv.output_encoding is deprecated in .../application/libraries/Engine/Application.php on line 315
Deprecated: iconv_set_encoding(): Use of iconv.internal_encoding is deprecated in .../application/libraries/Engine/Application.php on line 316
Deprecated: iconv_set_encoding(): Use of iconv.input_encoding is deprecated in .../application/modules/Core/Bootstrap.php on line 33
Deprecated: iconv_set_encoding(): Use of iconv.output_encoding is deprecated in .../application/modules/Core/Bootstrap.php on line 34
Deprecated: iconv_set_encoding(): Use of iconv.internal_encoding is deprecated in .../application/modules/Core/Bootstrap.php on line 35
We're sorry! We are currently experiencing some technical issues. Please try again later.
```
iconv.internal_encoding is [deprecated in PHP 5.6](http://php.net/manual/en/iconv.configuration.php).

**Comment by shivani2910** [comment]
Hi Robert,

Such deprecated warnings are being thrown because of PHP's latest version on your server. So, please check doing below changes in "/application/index.php" file and that will resolve this issue on your website:

Please search for below code:
```php
error_reporting(E_ALL);
```
and replace the above code with below one:
```php
error_reporting(E_ALL ^ E_DEPRECATED);
```

We hope your problem will be resolved doing this. Also, we've forwarded this issue to our development team so that you do not face this again in future after upgrading SocialEngine.

Best Regards,
Shivani

**Comment by shivani2910** [comment]
fixed in [4.8.8] (http://blog.socialengine.com/2015/03/23/socialengine-php-4-8-8-released/)

**Comment by jdouk69** [comment]
i have 4.9 and my error logs are being filled like cazy with this error.
i have tried the above fix with no luck.

**Comment by smazurov** [comment]
@jdouk69 its impossible for you to have 4.9 as we haven't released that version. If you have 4.9, then you either downloaded it from an illegal source, or you're not using SocialEngine PHP.

**Comment by jdouk69** [comment]
Yes I'm sorry it's 4.8.9

**Comment by Fendt1** [comment]
Dears,

we have this message since 4 days after we went to a new and faster server. But it shows only a white window. this is the output:

Deprecated: iconv_set_encoding(): Use of iconv.input_encoding is deprecated in /is/htdocs/wp12490775_3LF7W648U1/www/weet-weet_se/application/libraries/Engine/Application.php on line 314

Deprecated: iconv_set_encoding(): Use of iconv.output_encoding is deprecated in /is/htdocs/wp12490775_3LF7W648U1/www/weet-weet_se/application/libraries/Engine/Application.php on line 315

Deprecated: iconv_set_encoding(): Use of iconv.internal_encoding is deprecated in /is/htdocs/wp12490775_3LF7W648U1/www/weet-weet_se/application/libraries/Engine/Application.php on line 316

Deprecated: iconv_set_encoding(): Use of iconv.input_encoding is deprecated in /is/htdocs/wp12490775_3LF7W648U1/www/weet-weet_se/application/modules/Core/Bootstrap.php on line 33

Deprecated: iconv_set_encoding(): Use of iconv.output_encoding is deprecated in /is/htdocs/wp12490775_3LF7W648U1/www/weet-weet_se/application/modules/Core/Bootstrap.php on line 34

Deprecated: iconv_set_encoding(): Use of iconv.internal_encoding is deprecated in /is/htdocs/wp12490775_3LF7W648U1/www/weet-weet_se/application/modules/Core/Bootstrap.php on line 35

The Version is 4.89.

What can I do on this?
In the index.php was already written  this solution above error_reporting(E_ALL ^ E_DEPRECATED);

Bye
Fendt

**Comment by shivani2910** [comment]
Hi @Fendt1,

We're sorry for the inconvenience caused. I checked the support ticket you've filed with us and found that our support team member has replied to you regarding this issue. 
Please drop us a line if you still have any issues regarding this.

Thanks,
Shivani

**Comment by Fendt1** [comment]
Dear Shivani,

unfortunaltely HostEurope will not fix it. And with their suggestions it is
not working. They say it is old code and so they will not change more than
is possible by the user in the Adminpanel.

So we have to look for a new hoster.

Your company says TMDHosting is special for SE.

Whats about that it will run without these problems we had at HostEurope -
and that it is really faster than on our old Test-Server with weet-weet.com?

sincerly

Josef

2015-12-09 8:27 GMT+01:00 Shivani Sharma <notifications@github.com>:

**Comment by albertpak** [comment]
👍  @shivani2910 's fix worked - running socialengine on 5.6 , and encountered this error when switched over to dev mode.

**Comment by smazurov** [comment]
This has been addressed in [4.8.11](http://blog.socialengine.com/2016/05/26/socialengine-php-4-8-11-is-released/)

**Comment by viveksharma01** [comment]
I am using SE 4.8.12 and getting iconv.internal_encoding is deprecated error only when I try to access payment settings. I am using Google Cloud Platform to host SE. What should I do get rid of this issue. I am not able to run any pages related with payments. Please suggeset me solution for this issue.

**Comment by Khushboo016** [comment]
Hi @viveksharma01, Issue related to iconv.internal_encoding deprecation has been fixed in SocialEngine 4.8.9, but if you are still facing this on your site, then we suggest you to submit a support ticket for this issue from your SocialEngine account, so we can check this issue on your site. It might be due to some server configurations.

Thanks!
